src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,127 @@
+    void register_nullUser_notOk() {
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(null),
+                String.format("%s should be thrown for user is null", EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for login is null", EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        registrationService.register(user);
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for already existing login: "
+                        + DEFAULT_LOGIN, EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for password is null", EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_invalidMinPasswordLength_notOk() {
+        user.setPassword("abcd");
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for invalid password: "
+                        + user.getPassword(), EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_invalidMaxPasswordLength_notOk() {
+        user.setPassword("abcdefghijklmnopqrstuvwxyzabc");
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for invalid password: "
+                        + user.getPassword(), EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for age is null", EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_invalidMinAge_notOk() {
+        user.setAge(9);
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for a invalid age: "
+                        + user.getAge(), EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void register_invalidMaxAge_notOK() {
+        user.setAge(91);
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(user),
+                String.format("%s should be thrown for a invalid age: "
+                + user.getAge(), EXCEPTION_MESSAGE));
+    }
+
```

Comment:
you should add more OK tests, for every field and their requirements
for example - registerAge18_OK

Reply:
:white_check_mark: